### DISCUSSION TITLE
Compiling ZXing-CPP into a .so file

### DISCUSSION THREAD

**Discussion by DevChagrins** [zxing-cpp/zxing-cpp#546] [Q&A]
I'm working on a project that I want to also run on Android. I was looking to compile it into a .so to link against, as to avoid having to write extra code to pull in functionality through Java.

I'm not super familiar with how CMake works, and any help would be appreciated on how I could do this on a Windows machine.

**Comment by axxel** [comment]
I have not done that but I would believe that simply following the steps from the [README](https://github.com/zxing-cpp/zxing-cpp/blob/master/wrappers/android/README.md) also works on Windows. There is no need to get in touch with cmake directly in this case.

**Reply by DevChagrins** [reply]
That works for the Android wrapper, but I need a compiled version of ZXing-CPP in either a .a or .so format. But I'll dig through the gradle build files here and see if I cannot figure out something.
